File: system-test/typescript/src/testkit/dsl/port/given/steps/given-clock.ts
import type { GivenStep } from './given-step.js';

export interface GivenClock extends GivenStep {
  withTime(): GivenClock;
  withTime(time: string): GivenClock;
  withWeekday(): GivenClock;
  withWeekend(): GivenClock;
}

File: system/multitier/backend-clean-java/src/testSupport/java/com/mycompany/myshop/backend/testkit/dsl/port/given/steps/GivenClock.java
package com.mycompany.myshop.backend.testkit.dsl.port.given.steps;

import com.mycompany.myshop.backend.testkit.dsl.port.given.steps.base.GivenStep;

public interface GivenClock extends GivenStep {
    GivenClock withTime();

    GivenClock withTime(String time);
}
